doc/developer-notes.md

```diff
@@ -138,6 +141,22 @@ public:
 } // namespace foo
 ```
 
+Coding Style (C++ functions and methods)
+--------------------
+
+- When ordering function parameters, place input parameters first, then any
+  in-out parameters, followed by any output parameters.
+
+- *Rationale*: API consistency.
+
+- Prefer returning values directly to using in-out or output parameters. Use
+  `std::optional` where helpful for returning values (as well as for optional
```

Comment:
Good point. I think using `std::optional` *for optional inputs* (instead some magic default value, if there is no real default) has enough merit to be called out separately.

Reply:
> has enough merit to be called out separately.

Suggesting splitting it out to a separate bullet point?

Reply:
I do think that's clearer.

Reply:
Done.